common-lib/src/main/scala/com/gu/mediaservice/lib/config/MetadataConfig.scala

```diff
@@ -8,38 +8,39 @@ import Scalaz._
 
 import com.gu.mediaservice.model.{StaffPhotographer, ContractPhotographer, Photographer}
 
+case class KnownPhotographer(name: String, publication: String)
+
 object PhotographersList {
-  type Store = Map[String, String]
   type CreditBylineMap = Map[String, List[String]]
 
-  import MetadataConfig.{ staffPhotographers, contractedPhotographers }
+  import MetadataConfig.{ Store, staffPhotographers, contractedPhotographers }
 
   def creditBylineMap(store: Store): CreditBylineMap = store
-      .groupBy{ case (photographer, publication) => publication }
-      .map{ case (publication, photographers) => publication -> photographers.keys.toList.sortWith(_.toLowerCase < _.toLowerCase) }
-
-  def creditBylineMap(stores: List[Store]): CreditBylineMap =
-    stores.map(creditBylineMap).reduceLeft(_ |+| _)
+      .groupBy{ case KnownPhotographer(_, publication) => publication }
+      .map{ case (publication, photographers) => publication -> photographers.map(_.name).sortWith(_.toLowerCase < _.toLowerCase) }
 
-  def list(store: Store) = store.keys.toList.sortWith(_.toLowerCase < _.toLowerCase)
-
-  def getPublication(store: Store, name: String): Option[String] = store.get(name)
+  def list(store: Store) = store.map(_.name).sortWith(_.toLowerCase < _.toLowerCase)
 
   def caseInsensitiveLookup(store: Store, lookup: String) =
-    store.find{case (name, pub) => name.toLowerCase == lookup.toLowerCase}
+    store.reverse.find{case KnownPhotographer(name, _) => name.toLowerCase == lookup.toLowerCase}
 
   def getPhotographer(photographer: String): Option[Photographer] = {
     caseInsensitiveLookup(staffPhotographers, photographer).map {
-      case (name, pub) => StaffPhotographer(name, pub)
+      case KnownPhotographer(name, publication) => StaffPhotographer(name, publication)
     }.orElse(caseInsensitiveLookup(contractedPhotographers, photographer).map {
-      case (name, pub) => ContractPhotographer(name, Some(pub))
+      case KnownPhotographer(name, publication) => ContractPhotographer(name, Some(publication))
     })
   }
 }
 
 object MetadataConfig {
 
-  val externalStaffPhotographers: Map[String, String] = Map(
+  type Store = List[KnownPhotographer] // not a Map at this point to allow for duplicate keys (as some photographers take pics for multiple publications)
+
+  import scala.language.implicitConversions
+  implicit def toKnownPhotographer(value: (String, String)) = KnownPhotographer(value._1, value._2)
```

Comment:
Be lovely to match on this so we can name values 1 through 2 👍

Reply:
i went to go do that and I wonder if this is niiiice or confusing
```suggestion
  implicit def toKnownPhotographer(value: (String, String)) = KnownPhotographer.apply _
```

Reply:
i just wonder if `match` is too verbose
```suggestion
  implicit def toKnownPhotographer(value: (String, String)) = value match {
    case (name, publication) => KnownPhotographer(name, publication)
  }
```

Reply:
yeah, I guess the fact that this is implicit (yeesh) makes this less troubling – it'd be nice if you could get the semantics from the call sig, not the implementation, but with implicits I guess the idea is that you're not actually calling the thing yourself. Weird.

I'd favour the verbosity over scala-neatness, but it does seem reasonable to go for either.

Reply:
This might be an alternative:
```scala
  implicit class Photographer(name: String) {
    def ->(publication: String): KnownPhotographer = KnownPhotographer(name, publication)
  }
```

Reply:
ahh nice, I've gone for the `implicit class` approach, but called `KnownPhotographerOps` rather than `Photographer` (as that is already a thing, `com.gu.mediaservice.model.Photographer` inside `UsageRights.scala`).